kafkakewl-common/src/main/scala/com/mwam/kafkakewl/common/kafka/KafkaConsumerUtils.scala

```diff
@@ -86,22 +86,22 @@ object KafkaConsumerUtils {
     val subscription = Manual(topicPartitions.toSet)
     for {
       endOffsets <- consumer.endOffsets(topicPartitions.toSet)
+      // This will ensure that the topic stops once the messages up to the offsets
+      // of endOffsets have been read. Also prevents the topic consumer going into
+      // an infinite loop when there is one message.
+      stopFibre <- {
+        ZStream
+          .repeat(())
+          .schedule(Schedule.fixed(Duration.fromSeconds(1)))
+          .mapZIO(_ => ZIO.foreachPar(topicPartitions)(tp => consumer.position(tp).map(pos => (tp, pos))).map(_.toMap).debug)
+          .takeUntil(nextOffsets => !endOffsets.exists { case (tp, endOffset) => nextOffsets.get(tp).exists(nextOffset => nextOffset < endOffset) })
+          .run(ZSink.drain) *> consumer.stopConsumption
+      }.fork
       partitionMapAndRecords <- consumer
         .plainStream(subscription, keyDeserializer, valueDeserializer)
         // Batch up values from the consumer
         .aggregateAsyncWithin(ZSink.collectAll, Schedule.fixed(pollTimeout))
```

Comment:
Do we still need `aggregateAsyncWithin()` ? Ultimately we process each record one by one below, not sure how much it's better if they come in batches.